src/nvim/eval/typval_encode.c.h

```diff
@@ -351,8 +351,16 @@ static int TYPVAL_ENCODE_CONVERT_ONE_VALUE(
     break;
   case VAR_PARTIAL: {
     partial_T *const pt = tv->vval.v_partial;
-    (void)pt;
-    TYPVAL_ENCODE_CONV_FUNC_START(tv, (pt == NULL ? NULL : partial_name(pt)));
+    char *const fname = pt == NULL ? NULL : partial_name(pt);
+    char *ffname = NULL;
+    if (fname != NULL && pt->pt_name == NULL && isupper(fname[0])) {
```

Comment:
Should I use `mb_isupper()`? This header file is included several times so every header matters for code size.